### DISCUSSION TITLE
raw.githubusercontent.com broken?

### DISCUSSION THREAD

**Discussion by Prajyot-Parab** [community/community#32889] [Other Feature Feedback, Questions, & Ideas] [answered]
Facing issues with accessing `https://raw.githubusercontent.com` URLs ?

**Comment by airtower-luna** [comment, marked as answer]
Works just fine for me. You're more likely to get a useful answer if you describe in detail what kinds of problem you're seeing (URLs you want to access, commands you use, error messages you get, etc.).

**Reply by Prajyot-Parab** [reply]
Thanks for response @airtower-luna 

Example URL - [https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-ibmcloud/main/tilt-provider.json](https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-ibmcloud/main/tilt-provider.json), I am not able to access the same or any other random file too. Getting can't find server error.

**Reply by airtower-luna** [reply]
I'm getting a JSON file for that URL.

> Getting can't find server error.

That sounds like a DNS problem (but again, *exact and full* error message matters). If it's a DNS error you need to check your local network, or possibly with your provider. If the problem is a DNS server that's not working right using an alternative DNS service (for example [Quad9](https://quad9.net/)) might help you work around the problem.

**Reply by Prajyot-Parab** [reply]
@airtower-luna Here is the exact error - 
```
wget https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-ibmcloud/main/tilt-provider.json
--2022-09-13 17:07:43--  https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-ibmcloud/main/tilt-provider.json
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... failed: nodename nor servname provided, or not known.
wget: unable to resolve host address ‘raw.githubusercontent.com’
```

will try switching DNS to see if that helps (though never faced this issue earlier)

**Reply by airtower-luna** [reply]
Yep, that confirms it's a DNS problem. To figure out why you'll have to check the relevant DNS servers, common locations are:

* You own computer (if you run a local DNS caching server)
* The (W)LAN router
* Your provider (in many cases your local router forwards DNS requests to the provider)

Naturally I can't comment on those. If you're on a company or university network you might want to ask a network admin.

**Reply by Prajyot-Parab** [reply]
Adding alternative DNS worked. Thanks @airtower-luna

**Reply by JosephKJ** [reply]
Hi @Prajyot-Parab, what was the alternate DNS server that you added? I tried 8.8.8.8 and 8.8.4.4, but hitting the same issue.

**Reply by avishkarabhishek786** [reply]
> Hi @Prajyot-Parab, what was the alternate DNS server that you added? I tried 8.8.8.8 and 8.8.4.4, but hitting the same issue.

185.199.108.133 raw.githubusercontent.com

**Reply by MOHITSAINDANE** [reply]
in termux when me download Kali in termux this error show me solve my problem 😔😔😔😔😔😔😔😔😔. Connecting to raw.githubusercontent.com|2405:200:1607:2820:41:

:36:443... failed: Connection timed out.

Connecting to raw.githubusercontent.com|49-44-79-2361:443... f

ailed: Connection timed out.

Retrying. --2023-03-02 07:34:52- (try: 2) https://raw.githubuserconte nt.com/EXALAB/AnLinux-Resources/master/Scripts/Installer/Kali/

kali.sh

Connecting to raw.githubusercontent.com|2405:200:1607:2820:41: :36:443... failed: Connection timed out. Connecting to raw.githubusercontent.com | 49-44-79-2361-443... f ailed: Connection timed out. Retrying.

--2023-03-02 07:35:58-- (try: 3) https://raw.githubuserconte

nt.com/EXALAB/AnLinux-Resources/master/Scripts/Installer/Kali/

kali.sh Connecting to raw.githubusercontent.com|2405:200:1607:2820:41: :36:443... failed: Connection timed out. Connecting to raw.githubusercontent.com 49-44-79-2361-443... f

ailed: Connection timed out.

Retrying.

**Comment by devarsh-mavani-19** [comment]
I added alternative DNS. but no result. @Prajyot-Parab @airtower-luna

**Reply by Asta9900** [reply]
Check this? https://www.debugpoint.com/failed-connect-raw-githubusercontent-com-port-443/

**Reply by ashishredcarpet** [reply]
@Asta9900  thanks!

**Comment by jjmachan** [comment]
facing the same issue here. When I run host command I get 
```
> host raw.githubusercontent.com

raw.githubusercontent.com has address 49.44.79.236
raw.githubusercontent.com has IPv6 address 2405:200:1607:2820:41::36
```

when I use 
```
> dig 8.8.8.8 raw.githubusercontent.com

; <<>> DiG 9.18.1-1ubuntu1.2-Ubuntu <<>> 8.8.8.8 raw.githubusercontent.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 30998
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;8.8.8.8.                       IN      A

;; ANSWER SECTION:
8.8.8.8.                0       IN      A       8.8.8.8

;; Query time: 4 msec
;; SERVER: 127.0.0.53#53(127.0.0.53) (UDP)
;; WHEN: Mon Dec 05 19:25:51 IST 2022
;; MSG SIZE  rcvd: 52

;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 25674
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;raw.githubusercontent.com.     IN      A

;; ANSWER SECTION:
raw.githubusercontent.com. 33   IN      A       49.44.79.236

;; Query time: 0 msec
;; SERVER: 127.0.0.53#53(127.0.0.53) (UDP)
;; WHEN: Mon Dec 05 19:25:51 IST 2022
;; MSG SIZE  rcvd: 70
```

I get the same IP but that IP seem to be not responsive.

Any help?

**Reply by airtower-luna** [reply]
With `dig 8.8.8.8 raw.githubusercontent.com` you're not actually asking a different server, as the output shows: `SERVER: 127.0.0.53#53(127.0.0.53) (UDP)`

Try `dig @8.8.8.8 raw.githubusercontent.com` to use `8.8.8.8` as your DNS server.

**Reply by jjmachan** [reply]
yep that fixed that @airtower-luna, Thanks for your help 🙌🏽, changed the dns server to 8.8.8.8 to fix the issue

Seems like an issue with the default DNS server JIO Fiber is using for people facing the same issue in India. adding 8.8.8.8 and reconnecting to the wifi again fixed my issue

**Reply by rahulnag** [reply]
Yes its the issue with JIO Fiber only, i dont know whom to contact and how to resolve

**Reply by jsuryakt** [reply]
Same issue with Jio Fiber, added 8.8.8.8 to DNS list, and it worked
Thanks

**Reply by rahulsemwal** [reply]
Thanks everyone who made it possible to fix, it is really a disaster with Jio wifi connection specially in India which does not allow to get access to raw.githubusercontent.com website link. It is surely a DNS problem with Jio wifi connection and can be solved by adding 8.8.8.8 DNS (safe and secure DNS, recommended from google) in wifi DNS setting.

If someone do not know to add DNS in Mac OS so please refer https://support.apple.com/en-in/guide/mac-help/mh14127/mac#:~:text=To%20change%20these%20settings%2C%20choose,may%20need%20to%20scroll%20down.)&text=List%20of%20DNS%20servers.

**Reply by nitsuaj** [reply]
Thank you so much 😇

**Reply by andev3-printftech** [reply]
I am facing the same issue. By adding 8.8.8.8 to DNS list we can resolve issue only for our network. What for other users. Like if we are developing public website and app that uses raw.githubusercontent.com for static API and that websites and apps are uses all Indian audience? So we all need to tweet this issue in twitter withi jio care and jio reliance.

I have twitted with jio care and jio reliance. You all can find jio care from https://twitter.com/JioCare and jio reliance https://twitter.com/reliancejio

Also I have found some manager and heads of reliance jio on linked and I have personally messaged and mailed them regarding this issue.

Thanks & Regards

**Reply by athul7744** [reply]
@andev3-printftech thank you for reporting. I have been facing the same issue for the past week, unable to fetch https://raw.githubusercontent.com through Jio Fiber. Realised it when I was unable to load plugin data for LogSeq. It worked through VPN and by changing DNS to 1.1.1.1

**Reply by Jayakumar6** [reply]
> With `dig 8.8.8.8 raw.githubusercontent.com` you're not actually asking a different server, as the output shows: `SERVER: 127.0.0.53#53(127.0.0.53) (UDP)`
> 
> Try `dig @8.8.8.8 raw.githubusercontent.com` to use `8.8.8.8` as your DNS server.

how to change that dns server in my local machine

**Reply by kaushalgairola** [reply]
Thank you for solutions. 😊

**Reply by cibi-p** [reply]
Is there any way to report about this issue... I contacted jio support team, they said that this is GitHub problem...

**Reply by AnshVM** [reply]
It works when I use jio mobile data over hotspot, but not with Jio fiber

**Comment by ghost** [comment]
Have been making changes to files on github, some raw versions of the files that are large are not updating @ raw.githubusercontent.com. any idea why?

**Comment by ghost** [comment]
So delete a file on github why is the @ raw.githubusercontent.com still have the file.

**Comment by MOHITSAINDANE** [comment]
raw.githubusercontent.com|2405:200:1607:2820:41::36|:443... failed: Connection timed out.
Please solve my problem termux me

**Comment by theafolayan** [comment]
What fixed it for me was changing my DNS to google's: `8.8.8.8` and `8.8.4.4` saving changes and then restarting my mac. 

Everything worked fine after that.

**Reply by princebansal7** [reply]
Hey, I tried the same, but didn't work for me. Any alternative DNS ?

**Reply by airtower-luna** [reply]
You could try [Quad9](https://quad9.net/) (9.9.9.9 or 2620:fe::9).

**Reply by princebansal7** [reply]
Thanks, but didn't work!

**Comment by Abhinavkene** [comment]
The jio has block the server for some reason,  try to use a VPN then it will work for sure .
